website/content/posts/introducing-wash.md

```diff
@@ -0,0 +1,100 @@
+Console output of an EC2 instance           | `cat aws/foo/resources/ec2/instances/bar/console.out`
+Console output of a Google compute instance | `cat gcp/<project>/compute/foo/console.out`
+An S3 object's content                      | `cat aws/foo/resources/s3/bar/baz`
+A GCP Storage object's content              | `cat gcp/<project>/storage/foo/bar`
+
+<div style="width:200px">Exec `uname` on </div> | 
+-----------------------------|---
+An EC2 instance              | `wexec aws/foo/resources/ec2/instances/bar uname`
+An a Docker container        | `wexec docker/containers/foo uname`
+Exec on a Kubernetes pod     | `wexec kubernetes/<context>/<namespace>/pods/foo uname`
+On a Google Compute instance | `wexec gcp/<project>/compute/foo uname`
+
+<div style="width:200px">Find by 'owner' tag/label</div> | 
+------------------------|---
+EC2 instances           | `find aws/foo -k ‘*ec2*instance’ -meta ‘.tags[?].key’ owner`
+Docker containers       | `find docker -k ‘*container’ -meta ‘.labels.owner’ -exists`
+Kubernetes pods         | `find kubernetes -k ‘*pod’ -meta ‘.metadata.labels.owner’ -exists`
+Google Compute instance | `find gcp -k ‘*compute*instance’ -meta ‘.labels.owner’ -exists`
+
+<p style="text-align: center;">Table 2: Common CLI Tasks with Wash</p>
+
+Comparing Tables 1 and 2, we immediately see that using Wash means:
+
+* You no longer have to learn different commands to execute a task across different things. All you need is one command (find for List/Find; cat for Read; and wexec for Exec).
+
+* You no longer have to install a bunch of different tools. All you need to install is the Wash binary.
+
+* You no longer have to learn different DSLs for filtering stuff. All you need to learn is find’s expression syntax and its individual primaries. Once you do that, you can filter on almost any conceivable property of your specific thing.
+
+In fact, Wash is not just a centralization of basic and fundamental commands. It is a shell. It is a shell layered on top of an existing shell like Bash or ZSH. This means that:
+
+* “Everything is a file”. Thus, interacting with a specific kind of thing like a Docker container is as easy as cd’ing into its containing “directory” and running a bunch of commands on the things in that “directory”. For example, something like cd docker/containers lets you follow-up with commands like ls (list all containers), wexec foo uname (execute the uname command on the foo container), or tail foo/fs/var/log/messages (tail the foo container’s /var/log/messages file for updates).
+
+* You can tab-complete and glob stuff (if your shell supports those features). For example, assuming you’re in the docker/containers directory, typing in ls f then tab will work just the way you’d expect it to work if you were in a standard directory like /var/log (where f would be a regular file or directory). Similarly, tail */fs/var/log/messages will tail every container’s /var/log/messages file for updates.
```

Comment:
No footnote?

Reply:
Haven't figured out the formatting yet. Also not sure if it's necessary.

Reply:
Updated.

Reply:
Sweet. Small error in `tail find -k ‘aws/*ec2*instance’ -o -k ‘gcp/*compute*instance’ | sed s:$:/fs/var/log/messages:g`. This should be
```
tail `find -k 'aws/*ec2*instance' -o -k 'gcp/*compute*instance' | sed s:$:/fs/var/log/messages:g`
```

Reply:
Updated.